pkg/proxy/ipvs/meta_proxier.go

```diff
@@ -153,25 +153,47 @@ func (proxier *metaProxier) OnEndpointsSynced() {
 // OnEndpointSliceAdd is called whenever creation of a new endpoint slice object
 // is observed.
 func (proxier *metaProxier) OnEndpointSliceAdd(endpointSlice *discovery.EndpointSlice) {
-	// noop
+	switch endpointSlice.AddressType {
+	case discovery.AddressTypeIPv4:
+		proxier.ipv4Proxier.OnEndpointSliceAdd(endpointSlice)
+	case discovery.AddressTypeIPv6:
+		proxier.ipv6Proxier.OnEndpointSliceAdd(endpointSlice)
+	default:
```

Comment:
What's the reason for checking `discovery.AddressTypeIP` in the endpoints change tracker but not here? If we're supporting it even though it's deprecated I think we should we still check for it here. I guess we'd have to call both the ipv4 and ipv6 proxiers for that case.

Reply:
We're not supporting the creation of new EndpointSlices with the type IP in 1.17, but are allowing existing ones to continue to exist in a deprecated state. The EndpointSlice controller should convert any EndpointSlice it manages with the type IP to IPv4, but there is of course a chance that EndpointSlices are managed by other entities or that kube-proxy is upgraded to 1.17 before the controller. 

For a new feature like meta proxier with no previous support for the IP type, it didn't seem to make sense to support the IP address type. Of course it would also be a lot more complicated to support a type that could include both v4 and v6. Given that it was a deprecated alpha feature with no history of support for dual stack, it seemed fine to leave out here.

Reply:
Thanks for clarifying, makes sense!